h2o-py/h2o/frame.py

```diff
@@ -1865,11 +1865,35 @@ def merge(self, other, all_x=False, all_y=False, by_x=None, by_y=None, method="a
 
         :returns: New H2OFrame with the result of merging the current frame with the ``other`` frame.
         """
-        common_names = list(set(self.names) & set(other.names))
-        if not common_names:
-            raise H2OValueError("No columns in common to merge on!")
-        if by_x is None: by_x = [self.names.index(c) for c in common_names]
-        if by_y is None: by_y = [other.names.index(c) for c in common_names]
+        def getValidCols(by_idx, fr):  # so user can input names of the columns as well is idx num
```

Comment:
I would not define a new function here. This is more of a helper. You can add it under `Helpers` in `frame.py`